message Digit {
  required uint32 numeral = 1;
  repeated float data = 2;
}

message Features {
  required uint32 numeral = 1;
  required double x_feature = 2;
  required double y_feature = 3;
}

message Weights {
  repeated float weights = 1;
}

message PlotData {
  enum Type {
    point_type = 1;
    line_type = 2;
  }
  required Type type = 1;

  optional Features point = 2;
  optional Weights line  = 3;
}